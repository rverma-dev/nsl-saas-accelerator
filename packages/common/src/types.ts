import TIER from './stacks.json';

export interface DeploymentRecord {
  tenantId?: string; // callhealth
  id: string; // dev1, dev2, prod1, prod2
  type?: string; // can be pool or silo
  tier?: string; // small
  account?: string;
  region?: string;
}

export interface Deployment extends DeploymentRecord {
  provisioned?: boolean;
}
export function isValidTier(tierName: string): boolean {
  return TIER.some(tier => tier.name === tierName);
}

export function getPipelineName(record: DeploymentRecord): string {
  return `${record.tenantId}-${record.id}-${record.type}-pipeline`;
}
