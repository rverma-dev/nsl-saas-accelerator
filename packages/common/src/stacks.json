[
  {
    "name": "small",
    "eks": {
      "version": "1.25"
    },
    "msk": {
      "version": "2.8"
    },
    "rds": {
      "version": "10.x",
      "size": "db.small"
    },
    "dynamodb": {
      "size": "14-2"
    }
  },
  {
    "name": "small-rc",
    "eks": {
      "version": "1.25"
    },
    "msk": {
      "version": "2.8"
    },
    "rds": {
      "version": "10.x",
      "size": "db.small"
    },
    "dynamodb": {
      "size": "14-2"
    }
  },
  {
    "name": "large",
    "eks": {
      "version": "1.25"
    },
    "msk": {
      "version": "2.8"
    },
    "rds": {
      "version": "10.x",
      "size": "db.small"
    },
    "dynamodb": {
      "size": "100-80"
    }
  },
  {
    "name": "serverless",
    "eks": {
      "version": "1.25"
    },
    "msk": {
      "version": "2.8"
    },
    "rds": {
      "version": "10.x",
      "maxSize": "2"
    },
    "dynamodb": {
      "maxSize": "14"
    }
  }
]