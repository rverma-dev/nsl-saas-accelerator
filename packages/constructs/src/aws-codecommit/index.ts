/**
 *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */
import * as path from 'path';
import * as cdk from 'aws-cdk-lib';
import { RemovalPolicy } from 'aws-cdk-lib';
import { Construct } from 'constructs';

export interface CreateGitopsSecretProps {
  /**
   * Gitops IAM username
   */
  username: string;
  /**
   * Secret name
   */
  secretName: string;
}

/**
 * Class to configure CloudWatch Destination on logs receiving account
 */
export class CreateGitopsSecretResource extends Construct {
  readonly id: string;

  constructor(scope: Construct, id: string, props: CreateGitopsSecretProps) {
    super(scope, id);

    const RESOURCE_TYPE = 'Custom::CodeCommitCredentials';
    const accessRoleResourceName = 'CreateGitopsSecretLambda';
    const assetsAccessRole = new cdk.aws_iam.Role(this, accessRoleResourceName, {
      roleName: `AWSLambdaBasicExecutionRole-${this.node.id}`,
      assumedBy: new cdk.aws_iam.ServicePrincipal('lambda.amazonaws.com'),
      description: 'AWS Accelerator assets access role in workload accounts to bootstrap gitops credentials',
    });
    assetsAccessRole.addManagedPolicy(
      cdk.aws_iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
    );
    assetsAccessRole.addToPolicy(
      new cdk.aws_iam.PolicyStatement({
        resources: [`arn:${cdk.Aws.PARTITION}:logs:${cdk.Aws.REGION}:${cdk.Aws.ACCOUNT_ID}:*`],
        actions: ['logs:CreateLogGroup'],
      }),
    );
    assetsAccessRole.addToPolicy(
      new cdk.aws_iam.PolicyStatement({
        resources: [
          `arn:${cdk.Aws.PARTITION}:secretsmanager:${cdk.Aws.REGION}:${cdk.Aws.ACCOUNT_ID}:secret:${props.secretName}`,
        ],
        actions: ['secretsmanager:*'],
      }),
    );
    assetsAccessRole.addToPolicy(
      new cdk.aws_iam.PolicyStatement({
        resources: ['*'],
        actions: ['kms:Decrypt', 'kms:DescribeKey'],
      }),
    );
    assetsAccessRole.addToPolicy(
      new cdk.aws_iam.PolicyStatement({
        resources: [`arn:${cdk.Aws.PARTITION}:iam::${cdk.Aws.ACCOUNT_ID}:user/${props.username}`],
        actions: [
          'iam:CreateServiceSpecificCredential',
          'iam:DeleteServiceSpecificCredential',
          'iam:ListServiceSpecificCredentials',
        ],
      }),
    );
    //
    // Function definition for the custom resource
    //
    const providerLambda = new cdk.aws_lambda.Function(this, 'Function', {
      runtime: cdk.aws_lambda.Runtime.NODEJS_18_X,
      code: cdk.aws_lambda.Code.fromAsset(path.join(__dirname, '../../lib/aws-codecommit/lambda-codecommit-secret')),
      handler: 'index.handler',
      timeout: cdk.Duration.minutes(15),
      description: 'Create Gitops Secret',
      role: assetsAccessRole,
    });

    // Custom resource lambda log group
    new cdk.aws_logs.LogGroup(this, `${providerLambda.node.id}LogGroup`, {
      logGroupName: `/aws/lambda/${providerLambda.functionName}`,
      removalPolicy: RemovalPolicy.DESTROY,
    });

    const provider = new cdk.custom_resources.Provider(this, 'Custom::GitopsSecrets', {
      onEventHandler: providerLambda,
    });

    const resource = new cdk.CustomResource(this, 'Resource', {
      resourceType: RESOURCE_TYPE,
      serviceToken: provider.serviceToken,
      properties: {
        username: props.username,
        secretName: props.secretName,
      },
    });

    this.id = resource.ref;
  }
}
