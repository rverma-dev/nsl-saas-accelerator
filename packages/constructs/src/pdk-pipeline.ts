import { Aspects, RemovalPolicy, SecretValue, Stage } from 'aws-cdk-lib';
import { Pipeline } from 'aws-cdk-lib/aws-codepipeline';
import { GitHubTrigger } from 'aws-cdk-lib/aws-codepipeline-actions';
import { Key } from 'aws-cdk-lib/aws-kms';
import { BlockPublicAccess, Bucket, BucketEncryption, ObjectOwnership } from 'aws-cdk-lib/aws-s3';
import * as pipelines from 'aws-cdk-lib/pipelines';
import { NagSuppressions } from 'cdk-nag';
import { Construct } from 'constructs';

const DEFAULT_BRANCH_NAME = 'main';
const REPOSITORY_SECRET = 'saas-provisoner';

/**
 * Properties to configure the PDKPipeline.
 *
 * Note: Due to limitations with JSII and generic support it should be noted that
 * the synth, synthShellStepPartialProps.input and
 * synthShellStepPartialProps.primaryOutputDirectory properties will be ignored
 * if passed in to this construct.
 *
 * synthShellStepPartialProps.commands is marked as a required field, however
 * if you pass in [] the default commands of this construct will be retained.
 */
export interface PDKPipelineProps extends pipelines.CodePipelineProps {
  /**
   * Name of the CodeCommit repository to create.
   */
  readonly repositoryName: string;

  /**
   * Output directory for cdk synthesized artifacts i.e: packages/infra/cdk.out.
   */
  readonly primarySynthDirectory: string;

  /**
   * PDKPipeline by default assumes a NX Monorepo structure for it's codebase and
   * uses sane defaults for the install and run commands. To override these defaults
   * and/or provide additional inputs, specify env settings, etc you can provide
   * a partial ShellStepProps.
   */
  readonly synthShellStepPartialProps?: pipelines.ShellStepProps;

  /**
   * Branch to trigger the pipeline execution.
   *
   * @default mainline
   */
  readonly defaultBranchName?: string;

  /**
   * Alias to use for existing KMS Key when crossAccount.
   *
   * @default mainline
   */
  readonly existingKMSKeyAlias?: string;

  /**
   * Possible values for a resource's Removal Policy
   * The removal policy controls what happens to the resource if it stops being managed by CloudFormation.
   */
  readonly codeCommitRemovalPolicy?: RemovalPolicy;

  /**
   * If we want to reuse existing artifact bucket
   * However for cross account access its better to create bucket per tenant
   */
  readonly existingAccessLogBucket?: string;

  /**
   * If we want to reuse existing artifact bucket
   * However for cross account access its better to create bucket per tenant
   */
  readonly existingArtifactBucket?: string;

  /**
   * Determine if pipeline is for toolchain or workload
   */
  readonly isToolChain?: string;
}

/**
 * An extension to CodePipeline which configures sane defaults for a NX Monorepo
 * codebase. In addition to this, it also creates a CodeCommit repository with
 * automated PR builds and approvals.
 */
export class SaasPipeline extends Construct {
  readonly codePipeline: pipelines.CodePipeline;

  public constructor(scope: Construct, id: string, props: PDKPipelineProps) {
    super(scope, id);

    this.node.setContext('@aws-cdk/aws-s3:serverAccessLogsUseBucketPolicy', true);

    const accessLogsBucket = props.existingAccessLogBucket ?
      Bucket.fromBucketName(this, 'ExAccessLogsBucket', props.existingAccessLogBucket) :
      new Bucket(this, 'AccessLogsBucket', {
        versioned: false,
        enforceSSL: true,
        autoDeleteObjects: true,
        removalPolicy: RemovalPolicy.DESTROY,
        encryption: BucketEncryption.S3_MANAGED,
        objectOwnership: ObjectOwnership.BUCKET_OWNER_ENFORCED,
        publicReadAccess: false,
        blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
      });

    const artifactBucket = props.existingArtifactBucket ?
      Bucket.fromBucketName(this, 'ExArtifactsBucket', props.existingArtifactBucket) :
      new Bucket(this, 'ArtifactsBucket', {
        enforceSSL: true,
        autoDeleteObjects: true,
        removalPolicy: RemovalPolicy.DESTROY,
        encryption: props.crossAccountKeys ? BucketEncryption.KMS : BucketEncryption.S3_MANAGED,
        encryptionKey: props.crossAccountKeys && props.existingKMSKeyAlias ?
          Key.fromLookup(this, 'ArtifactKey', { aliasName: props.existingKMSKeyAlias }) :
          props.crossAccountKeys ?
            new Key(this, 'ArtifactKey', {
              enableKeyRotation: true,
              removalPolicy: RemovalPolicy.DESTROY,
            }) : undefined,
        objectOwnership: ObjectOwnership.BUCKET_OWNER_ENFORCED,
        publicReadAccess: false,
        blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
        serverAccessLogsPrefix: 'access-logs',
        serverAccessLogsBucket: accessLogsBucket,
      });

    const githubInput = pipelines.CodePipelineSource.gitHub(
      props.repositoryName,
      props.defaultBranchName || DEFAULT_BRANCH_NAME,
      {
        trigger: GitHubTrigger.NONE,
        authentication: SecretValue.secretsManager(REPOSITORY_SECRET, {
          jsonField: 'github_token',
        }),
      },
    );

    const codePipeline = new Pipeline(this, 'CodePipeline', {
      enableKeyRotation: props.crossAccountKeys,
      restartExecutionOnUpdate: true,
      crossAccountKeys: props.crossAccountKeys,
      artifactBucket,
    });

    // ignore input and primaryOutputDirectory
    const { input, primaryOutputDirectory, commands, ...synthShellStepPartialProps } =
    props.synthShellStepPartialProps || {};

    const synthShellStep = new pipelines.ShellStep('Synth', {
      input: githubInput,
      installCommands: ['npm install -g aws-cdk pnpm', 'pnpm install --frozen-lockfile'],
      commands: commands && commands.length > 0 ? commands : ['npx nx run-many --target=build --all'],
      primaryOutputDirectory: props.primarySynthDirectory,
      ...(synthShellStepPartialProps || {}),
    });

    synthShellStep.addOutputDirectory('.');

    const codePipelineProps: pipelines.CodePipelineProps = {
      codePipeline,
      ...props,
      crossAccountKeys: undefined,
      synth: synthShellStep,
    };

    this.codePipeline = new pipelines.CodePipeline(this, id, codePipelineProps);
  }

  /**
   * @inheritDoc
   */
  addStage(stage: Stage, options?: pipelines.AddStageOpts): pipelines.StageDeployment {
    // Add any root Aspects to the stage level as currently this doesn't happen automatically
    Aspects.of(stage.node.root).all.forEach(aspect => Aspects.of(stage).add(aspect));
    return this.codePipeline.addStage(stage, options);
  }

  addWave(wave: string, options?: pipelines.WaveOptions): pipelines.Wave {
    return this.codePipeline.addWave(wave, options);
  }

  buildPipeline() {
    this.codePipeline.buildPipeline();
    this.suppressCDKViolations();
  }

  suppressCDKViolations() {
    this.suppressRules(
      ['AwsSolutions-IAM5', 'AwsPrototyping-IAMNoWildcardPermissions'],
      'Wildcards are needed for dynamically created resources.',
    );

    this.suppressRules(
      ['AwsSolutions-CB4', 'AwsPrototyping-CodeBuildProjectKMSEncryptedArtifacts'],
      'Encryption of Codebuild is not required.',
    );

    this.suppressRules(
      ['AwsSolutions-S1', 'AwsPrototyping-S3BucketLoggingEnabled'],
      'Access Log buckets should not have s3 bucket logging',
    );
  }

  private suppressRules(rules: string[], reason: string) {
    NagSuppressions.addResourceSuppressions(
      this,
      rules.map(r => ({
        id: r,
        reason,
      })),
      true,
    );
  }
}
