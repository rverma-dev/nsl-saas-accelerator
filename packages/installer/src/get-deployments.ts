import { getCloudFormationStacks, getRegions, scanDynamoDB } from './lib/apitools';
import { saveConfig, isValidDeploymentRecord } from './lib/configtools';
import { DeploymentRecord, Deployment, getPipelineName } from '@nsa/common';

/*
 * This utility reads a point-in-time snapshot of the deployment database,
 * validates all records, and checks the provisioning status of each
 * deployment. The resulting file is stored in build_output directory,
 * so that it can be read by the update-deployments utility.
 *
 * Both the records read from deployment database, and the validated
 * output of the utility are logged, so that a clear record exists
 * of the input and output of the utility.
 */

processDeployments().catch(error => {
  console.error(error);
  process.exit(1);
});

async function processDeployments(): Promise<void> {
  const stacks = await getCloudFormationStacks();
  const regions = await getRegions();
  const records = await scanDynamoDB();
  verifyAndSaveData(records, regions, stacks);
}

function verifyAndSaveData(records: Array<DeploymentRecord>, regions: Array<string>, stacks: Array<string>): void {
  console.log('Records from deployment database:');
  console.log(records);

  const deployments: Array<Deployment> = [];

  records.forEach(record => {
    // Verify records are properly formatted
    try {
      if (isValidDeploymentRecord(record, regions)) {
        console.log(record);
        const deployment = record as Deployment;
        deployment.provisioned = stacks.includes(getPipelineName(record));
        deployments.push(deployment);
      }
    } catch (error) {
      console.error(`Deployment record ${record.tenantId} failed validation: ${error}. Ignoring record.`);
    }
  });

  console.log('Validated records:');
  console.log(deployments);

  saveConfig(deployments, 'deployments.json');
}
