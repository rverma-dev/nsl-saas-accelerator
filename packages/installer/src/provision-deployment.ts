import { execSync } from 'child_process';
import * as console from 'console';
import * as process from 'process';
import { DeploymentRecord, getPipelineName } from '@nsa/common';
import { getRegions } from './lib/apitools';
import { isValidDeploymentRecord } from './lib/configtools';

/*
 * This utility is responsible for provisioning a new tenant pipeline. The utility
 * is executed in the AWS CodeBuild project 'provisioning-project'.
 *
 * The provisioning-project is automatically invoked whenever a new deployment
 * record is inserted into the deployment database (DynamoDB table). DynamoDB
 * Streams is configured with a AWS Lambda Trigger, which will pass the attributes
 * of the new record to this utility using environment variables.
 *
 * DEPLOYMENT_TENANT_ID = Name of the tenant
 * DEPLOYMENT_ID     = ID of the deployment
 * DEPLOYMENT_TYPE   = Type of deployment (silo or pool)
 * COMPONENT_ACCOUNT = AWS Account ID for component resources for this deployment
 * COMPONENT_REGION  = AWS Region, as above
 */

if (process.env.DEPLOYMENT_TENANT_ID === undefined) {
  throw new Error('Missing required env variable DEPLOYMENT_TENANT_ID');
}
if (process.env.DEPLOYMENT_ID === undefined) {
  throw new Error('Missing required env variable DEPLOYMENT_ID');
}
if (process.env.DEPLOYMENT_TYPE === undefined) {
  throw new Error('missing required env variable DEPLOYMENT_TYPE');
}
if (process.env.DEPLOYMENT_TIER === undefined) {
  throw new Error('Missing workspace env variable DEPLOYMENT_TIER');
}
if (process.env.COMPONENT_ACCOUNT === undefined) {
  throw new Error('Missing required env variable COMPONENT_ACCOUNT');
}
if (process.env.COMPONENT_REGION === undefined) {
  throw new Error('Missing required env variable COMPONENT_REGION');
}

const newRecord: DeploymentRecord = {
  tenantId: process.env.DEPLOYMENT_TENANT_ID,
  id: process.env.DEPLOYMENT_ID,
  type: process.env.DEPLOYMENT_TYPE,
  tier: process.env.DEPLOYMENT_TIER,
  account: process.env.COMPONENT_ACCOUNT,
  region: process.env.COMPONENT_REGION,
};

console.log('New deployment record:');
console.log(newRecord);

// Validate the record, and provision new tenant if validation passes
processRecord(newRecord).catch(error => {
  console.error(error);
  process.exit(1);
});

async function processRecord(record: DeploymentRecord): Promise<void> {
  const regions = await getRegions();
  if (isValidDeploymentRecord(record, regions)) {
    console.log(`Provisioning new deployment ${record.tenantId}`);
    await provisionPipelineStack(record);
  }
}

// Provision new silo or pool deployment. We will call CDK deploy with specific runtime context values.
async function provisionPipelineStack(record: DeploymentRecord): Promise<void> {
  const stackName = getPipelineName(record);
  const command =
    `yarn cdk deploy ${stackName} --toolkit-stack-name nsl-CDKToolkit --require-approval never` +
    ` -c deployment_id=${record.id}` +
    ` -c tenant_id=${record.tenantId}` +
    ` -c deployment_type=${record.type}` +
    ` -c deployment_tier=${record.tier}` +
    ` -c component_account=${record.account}` +
    ` -c component_region=${record.region} `;

  console.log('Executing: ' + command);
  execSync(command, { stdio: 'inherit' });
}
