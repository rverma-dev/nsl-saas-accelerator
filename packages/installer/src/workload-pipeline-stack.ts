import { DemoPipeline } from '@nsa/demo/lib/pipeline-stack';
import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { DeploymentRecord, getPipelineName } from './lib/types';
export interface WorkloadPipelineProps extends StackProps, DeploymentRecord {}

export class WorkloadPipelineStack extends Stack {
  constructor(scope: Construct, id: string, props: WorkloadPipelineProps) {
    super(scope, id, props);

    // const ecrRepo = aws_ecr.Repository.fromRepositoryName(this, 'get-installer-repo', 'nsa-installer');
    // const sourceInput = pipelines.CodePipelineSource.gitHub(`${REPOSITORY_OWNER}/${REPOSITORY_NAME}`, 'main', {
    //   trigger: GitHubTrigger.NONE,
    //   authentication: SecretValue.secretsManager(REPOSITORY_SECRET, {
    //     jsonField: 'github_token',
    //   }),
    // });
    const pipelineName = getPipelineName(props);
    new DemoPipeline(this, pipelineName, {
      tenantID: props.tenantId!,
      deploymentId: props.id,
      deploymentType: props.type!,
      deploymentTier: props.tier!,
      env: { account: props.account, region: props.region }, // defines where the resources will be provisioned
    });
    // new pipelines.CodeBuildStep('synth', {
    //   input: sourceInput,
    //   buildEnvironment: {
    //     buildImage: aws_codebuild.LinuxArmBuildImage.fromEcrRepository(ecrRepo),
    //     computeType: ComputeType.SMALL,
    //   },
    //   commands: [
    //     'cd /app',
    //     `yarn cdk synth --toolkit-stack-name nsl-CDKToolkit -q --verbose \
    //     -c tenant_id=${props.tenantId} -c deployment_type=${props.type} -c deployment_id=${props.id} \
    //     -c component_account=${props.account} -c deployment_tier=${props.tier} -c component_region=${props.region} `,
    //   ],
    //   primaryOutputDirectory: '/app/cdk.out',
    // });
    //
    //
    // const pipeline = new pipelines.CodePipeline(this, pipelineName, {
    //   pipelineName: pipelineName,
    //   selfMutation: true,
    //   synth: synthStep,
    //   crossAccountKeys: true,
    //   cliVersion: CDK_VERSION,
    // });
    // pipeline.addStage(
    //   new DemoSite(this, props.id, {
    //     tenantID: props.tenantId!,
    //     deploymentId: props.id,
    //     deploymentType: props.type!,
    //     deploymentTier: props.tier!,
    //     env: { account: props.account, region: props.region }, // defines where the resources will be provisioned
    //   }),
    // );
  }
}
