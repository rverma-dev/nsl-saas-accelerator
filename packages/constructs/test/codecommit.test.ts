/**
 *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as cdk from 'aws-cdk-lib';
import { snapShotTest } from './snapshot-test';
import { CreateGitopsSecretResource } from '../src';

const testNamePrefix = 'Construct(GitopsStack): ';

//Initialize stack for snapshot test and resource configuration test
const stack = new cdk.Stack();
describe('GitopsSecret', () => {
  new CreateGitopsSecretResource(stack, 'codeCommitSecret', {
    username: 'demo',
    secretName: 'demo',
  });
  snapShotTest(testNamePrefix, stack);
});
