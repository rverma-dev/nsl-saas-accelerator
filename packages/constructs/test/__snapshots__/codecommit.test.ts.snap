// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitopsSecret Construct(GitopsStack):  Snapshot Test 1`] = `
{
  "Resources": {
    "codeCommitSecretCreateGitopsSecretLambdaD568B560": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Description": "AWS Accelerator assets access role in workload accounts to bootstrap gitops credentials",
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "RoleName": "AWSLambdaBasicExecutionRole-codeCommitSecret",
      },
      "Type": "AWS::IAM::Role",
    },
    "codeCommitSecretCreateGitopsSecretLambdaDefaultPolicyE6BF4509": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "logs:CreateLogGroup",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":logs:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":*",
                  ],
                ],
              },
            },
            {
              "Action": "secretsmanager:*",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":secretsmanager:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":secret:demo",
                  ],
                ],
              },
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": [
                "iam:CreateServiceSpecificCredential",
                "iam:DeleteServiceSpecificCredential",
                "iam:ListServiceSpecificCredentials",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":iam::",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":user/demo",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "codeCommitSecretCreateGitopsSecretLambdaDefaultPolicyE6BF4509",
        "Roles": [
          {
            "Ref": "codeCommitSecretCreateGitopsSecretLambdaD568B560",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "codeCommitSecretCustomGitopsSecretsframeworkonEventC5B2CBCC": {
      "DependsOn": [
        "codeCommitSecretCustomGitopsSecretsframeworkonEventServiceRoleDefaultPolicy4C82C9C8",
        "codeCommitSecretCustomGitopsSecretsframeworkonEventServiceRoleEE534113",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "REPLACED-GENERATED-NAME.zip",
        },
        "Description": "AWS CDK resource provider framework - onEvent (Default/codeCommitSecret/Custom::GitopsSecrets)",
        "Environment": {
          "Variables": {
            "USER_ON_EVENT_FUNCTION_ARN": {
              "Fn::GetAtt": [
                "codeCommitSecretFunction749A4ACD",
                "Arn",
              ],
            },
          },
        },
        "Handler": "framework.onEvent",
        "Role": {
          "Fn::GetAtt": [
            "codeCommitSecretCustomGitopsSecretsframeworkonEventServiceRoleEE534113",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "codeCommitSecretCustomGitopsSecretsframeworkonEventServiceRoleDefaultPolicy4C82C9C8": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "codeCommitSecretFunction749A4ACD",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "codeCommitSecretFunction749A4ACD",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "codeCommitSecretCustomGitopsSecretsframeworkonEventServiceRoleDefaultPolicy4C82C9C8",
        "Roles": [
          {
            "Ref": "codeCommitSecretCustomGitopsSecretsframeworkonEventServiceRoleEE534113",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "codeCommitSecretCustomGitopsSecretsframeworkonEventServiceRoleEE534113": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "codeCommitSecretF2AEED65": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "codeCommitSecretCustomGitopsSecretsframeworkonEventC5B2CBCC",
            "Arn",
          ],
        },
        "secretName": "demo",
        "username": "demo",
      },
      "Type": "Custom::CodeCommitCredentials",
      "UpdateReplacePolicy": "Delete",
    },
    "codeCommitSecretFunction749A4ACD": {
      "DependsOn": [
        "codeCommitSecretCreateGitopsSecretLambdaDefaultPolicyE6BF4509",
        "codeCommitSecretCreateGitopsSecretLambdaD568B560",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "REPLACED-GENERATED-NAME.zip",
        },
        "Description": "Create Gitops Secret",
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "codeCommitSecretCreateGitopsSecretLambdaD568B560",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "codeCommitSecretFunctionLogGroup8699C01D": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "LogGroupName": {
          "Fn::Join": [
            "",
            [
              "/aws/lambda/",
              {
                "Ref": "codeCommitSecretFunction749A4ACD",
              },
            ],
          ],
        },
        "RetentionInDays": 731,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
